softlayer/stemcell/softlayer_finder_test.go

```diff
@@ -26,23 +27,43 @@ var _ = Describe("SoftLayerFinder", func() {
 
 		logger = boshlog.NewLogger(boshlog.LevelNone)
 		finder = NewSoftLayerFinder(softLayerClient, logger)
+
+		expectedStemcell = NewSoftLayerStemcell(4868344, "8c7a8358-d9a9-4e4d-9345-6f637e10ccb7", VirtualDiskImageKind, softLayerClient, logger)
 	})
 
 	Describe("Find", func() {
-		Context("valid stemcell ID pointing to a SL virtual disk image", func() {
-			It("returns stemcell and found as true if stemcell ", func() {
+		Context("valid stemcell UUID pointing to a SL virtual disk image", func() {
+			It("returns stemcell and found as true if stemcell exists", func() {
 				stemcell, found, err := finder.Find("8c7a8358-d9a9-4e4d-9345-6f637e10ccb7")
 				Expect(err).ToNot(HaveOccurred())
 				Expect(found).To(BeTrue())
+				Expect(stemcell).To(Equal(expectedStemcell))
+			})
+		})
 
-				expectedStemcell := NewSoftLayerStemcell(4868344, "8c7a8358-d9a9-4e4d-9345-6f637e10ccb7", VirtualDiskImageKind, softLayerClient, logger)
+		Context("valid stemcell UUID pointing to a SL virtual disk image", func() {
+			It("returns found as false if stemcell does not exist", func() {
+				stemcell, found, err := finder.Find("fake-stemcell-uuid")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(found).To(BeFalse())
+				Expect(stemcell).To(BeNil())
+			})
+		})
+	})
+
+	Describe("FindById", func() {
+		Context("valid stemcell ID pointing to a SL virtual disk image", func() {
+			It("returns stemcell and found as true if stemcell ", func() {
+				stemcell, found, err := finder.FindById(4868344)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(found).To(BeTrue())
 				Expect(stemcell).To(Equal(expectedStemcell))
 			})
 		})
 
 		Context("valid stemcell ID pointing to a SL virtual disk image", func() {
-			It("returns found as false if stemcell directory does not exist", func() {
-				stemcell, found, err := finder.Find("fake-stemcell-id")
+			It("returns found as false if stemcell does not exist", func() {
+				stemcell, found, err := finder.FindById(1111111)
```

Comment:
No new line needed between `Expect` calls that are testing the same objects. Typically we just group related assertions.